### DISCUSSION TITLE
Faced training problem using paddleOCR for new Language(Bengali or Bangla)

### DISCUSSION THREAD

**Discussion by Hasibur013** [PaddlePaddle/PaddleOCR#15058] [Q&A]
I create a custom dataset using number plate image, text are bangla character and bangla number mixed, then I did rec_gt_train.txt, rec_gt_test.txt and word_dict_bn.txt file perfectly, then change yml file that need. After that I train this but got acc=0, tried different way still now acc=0. Did I any mistake? or any suggestions, please share your thought. If anyone did this for any new language please share your journey.

**Comment by GreatV** [comment]
You’re experiencing acc=0 when training PaddleOCR for Bangla number plates, despite preparing the dataset and config files. This is a common issue when training custom OCR for a new language. Let’s go through common mistakes and suggestions specific to your scenario:

### 1. **Check Your Dictionary File**
- Make sure your `word_dict_bn.txt` covers **every Bangla character and number** in your dataset.
- Each character or symbol (including spaces or special characters) present in your ground truth files must be included—one per line.

### 2. **Ground Truth File Format**
- Your `rec_gt_train.txt` and `rec_gt_test.txt` should follow PaddleOCR’s required format:
  ```
  path/to/image_1.jpg    corresponding_text_১
  path/to/image_2.jpg    কিছু_আরও_লেখা
  ```
- Ensure there are **no tab or space issues**; the separator is typically a tab or special separator.
- Check for **broken text encoding**. Bangla text must be UTF-8 encoded throughout.

### 3. **YAML Configuration**
- Verify you set the correct `character_dict_path` in the yaml config, pointing to your dictionary.
- Confirm `max_text_length` can accommodate the longest string in your dataset.

### 4. **Dataset Images**
- Make sure your images are **clear and labeled correctly**. Corrupted or empty labels can result in acc=0.
- Try training with a tiny subset (5–10 images) and validate it works, then scale up.

### 5. **Start with a Pretrained Model**
- Consider initializing training with a model pretrained on a similar script (like Hindi or Devanagari, or even default English), then fine-tune.

### 6. **Increase Dataset Size**
- If your dataset is too small or lacks variation, the model may not learn; try increasing dataset size and diversity.

### 7. **PaddleOCR Issues & Discussions**
Several users have faced similar issues when training on Bangla:
- [Bangla OCR Discussion - PaddleOCR GitHub](https://github.com/PaddlePaddle/PaddleOCR/discussions/14218): Multiple users successfully prepared Bangla datasets and offered tips on dictionary setup and config tweaks. Reading this may offer extra insight.

### 8. **Logging & Debugging**
- Run initial epochs with `print_batch_step: 1` in your config to observe what’s being fed into the model.
- Look for errors, warnings, or incorrectly formatted batches.

---

**Summary Checklist:**
- [ ] All unique characters in `word_dict_bn.txt` (check for missing or extra).
- [ ] Correct, UTF-8 encoded ground truth files formatted as required.
- [ ] YAML config paths and settings pointing to your files.
- [ ] Dataset images are readable and correctly labeled.
- [ ] Try starting with a small dataset and pretrained weights.

If you still see acc=0, consider sharing a snippet of your ground truth and dictionary file in the [Bangla OCR Discussion](https://github.com/PaddlePaddle/PaddleOCR/discussions/14218) for targeted help from other users who trained with new languages.

Good luck with your Bangla OCR project!

---
*Response generated by 🤖 [feifei-bot](https://github.com/GreatV/feifei) | gpt-4.1*

 ⚠️ This is an AI-generated response. Please verify any critical information.

**Reply by Hasibur013** [reply]
I followed all but still 0 acc

**Reply by cuicheng01** [reply]
Feel free to submit your request directly in this issue:
👉 [PaddleOCR Multilingual Support Request #15617](https://github.com/PaddlePaddle/PaddleOCR/issues/15617)
